src/CreeDictionary/API/search/presentation.py

```diff
@@ -19,6 +20,7 @@ class SerializedPresentationResult(TypedDict):
     wordform_text: str
     is_lemma: bool
     definitions: Iterable[SerializedDefinition]
+    lexical_information: List[Dict[str, Any]]
```

Comment:
Please avoid using the `Any` type. It defeats the purpose of using static type-checking :/

I believe the type you want represents this data, correct?

```python
  {
                    "entry": serialize_wordform(preverb_result),
                    "type": "Preverb",
                    "index": i,
                    "original_tag": tag,
                }
```

You could create a `TypedDict` for this or a dataclass:

```python
@dataclass
class SerializedMorpheme:
    entry: SerializedWordform
    type: Literal["Preverb", "Reduplication", "InitialChange"]
    index: int
    original_tag: FSTTag
```

I feel like this still could be iterated upon :/

Reply:
We should also be prepared for Prenouns, which are quite similar to Preverbs but attach instead to nouns (and are often the same as preverbs, but with an attributive rather than adverbial meaning).

Reply:
Noted. I'll play around with the types and dataclass idea and see what I can do.